command_parse.c

```diff
@@ -135,6 +135,24 @@ int parse_grouplist(struct GroupList *gl, struct Buffer *buf, struct Buffer *s,
   return 0;
 }
 
+enum CommandResult mutt_parse_rc_line_cwd(const char *line, struct Buffer *err, char *cwd) {
+    mutt_list_insert_head(&MuttrcStack, mutt_str_dup(NONULL(cwd)));
+
+    enum CommandResult ret = mutt_parse_rc_line(line, err);
+
+    struct ListNode *np = STAILQ_FIRST(&MuttrcStack);
+    STAILQ_REMOVE_HEAD(&MuttrcStack, entries);
+    FREE(&np->data);
+    FREE(&np);
+
+    return ret;
+}
+
+char *mutt_get_sourced_cwd(void) { // caller is responsible for freeing returned string
+    struct ListNode *np = STAILQ_FIRST(&MuttrcStack);
+    return np ? np->data ? mutt_str_dup(np->data) : NULL : NULL;
```

Comment:
If the `MuttrcStack` is empty, then we should add the current working dir.
My quick hack was:
```c
  struct Buffer *cwd = mutt_buffer_pool_get();
  mutt_path_getcwd(cwd);
  mutt_buffer_addstr(cwd, "/dummy.rc");
  char *dir = mutt_buffer_strdup(cwd);
  mutt_buffer_pool_release(&cwd);
  return dir;
```